src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -437,54 +432,6 @@ public void testProjectDeleteAfterSnapshotDelete() throws Exception {
         googleResourceManagerService.getProject(googleProjectId).getLifecycleState());
   }
 
-  @Ignore("Remove ignore after DR-1770 is addressed")
```

Comment:
I originally went to refactor this test so that the faults weren't needed.  But I saw that the referenced ticket was closed as "won't due" as the issue hadn't reoccurred in some time, so I deleted the test:

https://broadworkbench.atlassian.net/browse/DR-1770

Reply:
Thank you for probing this!

The reason the test issue had not reoccurred is because we [added the `@Ignore` annotation](https://github.com/DataBiosphere/jade-data-repo/commit/65400eefe70a486bc06f0f062c0c0bb191a5b1a5#diff-72cb89b8002a930037bb80cebaa142854e56c0da73e50468796fa1b5a7c99573).  We have not been running this test for several years without any ill effects, so one could argue that it may not have been valuable to us to begin with.

What was helpful in our post-standup conversation was to think through the question: "what behavior are we trying to verify with this test?".  That lead me to the initial [PR](https://github.com/DataBiosphere/jade-data-repo/pull/446) and [ticket](https://broadworkbench.atlassian.net/browse/DR-884).

The initial intent was to make sure that snapshot deletion takes out an exclusive lock on the snapshot, and that concurrent attempts at snapshot deletion would fail as a result.

A more complete, precise, stable, and speedy way to test this is through a chain of unit tests:
- Verify that `SnapshotDeleteFlight` adds `LockSnapshotStep` to its manifest in the way we expect
- Verify that `LockSnapshotStep` behaves as we'd expect for all possible input conditions (including if the snapshot is already locked, has already been deleted, etc).
- Verify that the underlying DAO methods to lock / unlock snapshots behave as we'd expect

We already had unit tests to verify DAO behavior.  I supplemented tests to include unit tests for flight construction and step behavior.  For symmetry, I also wrote unit tests for `UnlockSnapshotStep`, even though it's not called in this flight.

Reply:
That's great! Thanks for looking into this and writing tests that are more aligned with our current test strategy. I'm glad this allowed for removal of the fault insertion.

Reply:
Would it make sense to do this for `DatasetConnectedTest.testOverlappingDeletes` too? (This doesn't have to happen in this PR though)

Reply:
Good idea, thanks for flagging that! I'll take a look.

Reply:
I'm glad you flagged this :)

I was able to remove `DatasetConnectedTest.testOverlappingDeletes` by adding a unit test for `DatasetDeleteFlight` construction.  We already had unit tests to verify DAO behavior and step behavior.

But the fun part came when moving to remove the faults referenced in that connected test.  I found them referenced in `DatasetLockConnectedTest`, in tests that should have been failing with my code changes… but I realized that this test suite was not annotated and thus its tests weren't running with our connected tests as expected.  That's been the case for the last 2 years.

https://github.com/DataBiosphere/jade-data-repo/blob/cd8e294526a21a43aa74a6acb69b0b05115098d3/src/test/java/bio/terra/service/dataset/DatasetLockConnectedTest.java#L54

So I went through those connected tests, verified the behavior with additional flight construction unit tests where necessary, and got to remove 9 faults and clean up everywhere they were called in production code!